### PR TITLE
[DGPFUS-27] Develop an E-commerce Store Database

### PR DESCRIPTION
**Pull request by ayodejimb**
In this task, I designed a relational database for the FoodBag Store. Thank you.

### CODE REVIEW

File: submissions/mubarakOlaoluwa/Database/init.sql [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		   ('Vegetable & Fruits'),
+           ('Drinks & Cocktails'),
+           ('Resturants');
+
+/* This is the table for the second success criteria (vendors food) */
+CREATE TABLE vendor_food (
+	id INT NOT NULL AUTO_INCREMENT,
+    admin_id INT NOT NULL,
+    food_id INT NOT NULL,
+    name VARCHAR(100) NOT NULL,
```
Comment (ToluEnny24) [inline submissions/mubarakOlaoluwa/Database/init.sql:23]: let this accommodate more characters. 250 is fine

File: submissions/mubarakOlaoluwa/Database/init.sql [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    amount DECIMAL(10,2) NOT NULL,
+    PRIMARY KEY (id),
+    FOREIGN KEY (food_id) REFERENCES food_table(id)
```
Comment (ToluEnny24) [inline submissions/mubarakOlaoluwa/Database/init.sql:26]: the admin_id was not referenced as a foreign key

File: submissions/mubarakOlaoluwa/Database/init.sql [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+);
+
+/*This is the table for the third success criteria (admins table)*/
+CREATE TABLE admins(
+	id INT NOT NULL AUTO_INCREMENT,
+    name VARCHAR(50) NOT NULL,
```
Comment (ToluEnny24) [inline submissions/mubarakOlaoluwa/Database/init.sql:32]: let the characters be 250.

File: submissions/mubarakOlaoluwa/Database/init.sql [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    type INT NOT NULL,
+    email_address VARCHAR(50) NOT NULL,
```
Comment (ToluEnny24) [inline submissions/mubarakOlaoluwa/Database/init.sql:34]: This should accommodate more characters too

File: submissions/mubarakOlaoluwa/Database/init.sql [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    phone_number INT NOT NULL,
```
Comment (ToluEnny24) [inline submissions/mubarakOlaoluwa/Database/init.sql:35]: For a mobile number that starts with +2348078888, you will agree with me that, + wont be accepted as an integer or 08033003800, the 0 will be unaccepted too, likewise VARCHAR means capita letter O can be accepted as 0.

Just wanted you to understand the datatype you chose and what it entails

File: submissions/mubarakOlaoluwa/Database/init.sql [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    password VARCHAR(20) NOT NULL,
+    PRIMARY KEY (id)
+	);
+
+/*This is the table for the fourth success criteria (customer table)*/
+CREATE TABLE customer(
+	id INT NOT NULL AUTO_INCREMENT,
+    name VARCHAR(50) NOT NULL,
```
Comment (ToluEnny24) [inline submissions/mubarakOlaoluwa/Database/init.sql:43]: as usual

File: submissions/mubarakOlaoluwa/Database/init.sql [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    phone_number INT NOT NULL,
+    address VARCHAR(100) NOT NULL,
```
Comment (ToluEnny24) [inline submissions/mubarakOlaoluwa/Database/init.sql:45]: this should accommodate more characters

File: submissions/mubarakOlaoluwa/Database/init.sql [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    payment_method VARCHAR(20) NOT NULL,
```
Comment (ToluEnny24) [inline submissions/mubarakOlaoluwa/Database/init.sql:46]: This should be optional, we could have a first timer

File: submissions/mubarakOlaoluwa/Database/init.sql [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    card_number INT NOT NULL,
```
Comment (ToluEnny24) [inline submissions/mubarakOlaoluwa/Database/init.sql:47]: Not neccessary

File: submissions/mubarakOlaoluwa/Database/init.sql [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+    card_name VARCHAR (50) NOT NULL,
```
Comment (ToluEnny24) [inline submissions/mubarakOlaoluwa/Database/init.sql:48]: Removr the card details there are other payment methods

File: submissions/mubarakOlaoluwa/Database/init.sql [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+    expiry INT NOT NULL,
+    CVV INT NOT NULL,
+    PRIMARY KEY (id)
+	);
+    
+/*This is the table for the fifth success criteria (orders table)*/
+CREATE TABLE orders(
+	id INT NOT NULL AUTO_INCREMENT,
+    vendor_id INT NOT NULL,
+    customer_id INT NOT NULL,
```
Comment (ToluEnny24) [inline submissions/mubarakOlaoluwa/Database/init.sql:58]: this should be null since it is optional

**Review by ToluEnny24 — CHANGES_REQUESTED** [review]
@ayodejimb Well done please effect all requested changes.

**Comment by ayodejimb** [comment]
Thank you for those corrections. I have effected them all. Kindly check it out. Thanks.

**Pull request merged**